.travis.yml

```diff
@@ -17,50 +12,53 @@ script:
   - 'bundle exec rake $CHECK'
 bundler_args: 
 rvm:
-  - 2.5.1
-env:
-  global:
-    - BEAKER_PUPPET_COLLECTION=puppet6 PUPPET_GEM_VERSION="~> 6.0"
+  - 2.5.3
+stages:
+  - static
+  - spec
+  - acceptance
+  -
+    if: tag =~ ^v\d
+    name: deploy
 matrix:
   fast_finish: true
   include:
     -
+      bundler_args: --with system_tests
       dist: trusty
       env: PUPPET_INSTALL_TYPE=agent BEAKER_debug=true BEAKER_PUPPET_COLLECTION=puppet6 BEAKER_set=docker/centos-7 BEAKER_TESTMODE=apply
-      rvm: 2.5.1
+      rvm: 2.5.3
       script: bundle exec rake beaker
       services: docker
+      stage: acceptance
       sudo: required
     -
+      bundler_args: --with system_tests
       dist: trusty
       env: PUPPET_INSTALL_TYPE=agent BEAKER_debug=true BEAKER_PUPPET_COLLECTION=puppet6 BEAKER_set=docker/ubuntu-16.04 BEAKER_TESTMODE=apply
-      rvm: 2.5.1
+      rvm: 2.5.3
       script: bundle exec rake beaker
       services: docker
+      stage: acceptance
       sudo: required
     -
-      env: CHECK="syntax lint metadata_lint check:symlinks check:git_ignore check:dot_underscore check:test_file rubocop"
-    -
-      env: CHECK=parallel_spec
+      env: CHECK="check:symlinks check:git_ignore check:dot_underscore check:test_file rubocop syntax lint metadata_lint"
+      stage: static
     -
       env: PUPPET_GEM_VERSION="~> 5.0" CHECK=parallel_spec
-      rvm: 2.4.4
+      rvm: 2.4.5
+      stage: spec
+    -
+      env: PUPPET_GEM_VERSION="~> 6.0" CHECK=parallel_spec
+      rvm: 2.5.3
+      stage: spec
     -
-      env: PUPPET_GEM_VERSION="~> 4.0" CHECK=parallel_spec RUBYGEMS_VERSION=2.7.8
```

Comment:
This drops testing for Puppet 4.x. @suckatrash Any plans on what we will support for Puppet versions moving forward? Are we alright with dropping spec testing for 4.x?

Reply:
Can always add it back in with something like this if we want.  

https://github.com/npwalker/recursive_file_permissions/blob/master/.sync.yml#L6-L7

Reply:
I think we can drop Puppet 4 testing.  Voxpupuli started removing tests on Puppet 4 back in January, and the oldest supported PE version (2018.1) comes with Puppet 5.